reservation_units/models.py

```diff
@@ -685,6 +698,74 @@ def save(self, *args, **kwargs) -> None:
         if settings.UPDATE_PRODUCT_MAPPING:
             refresh_reservation_unit_product_mapping.delay(self.pk)
 
+    # ElasticSearch
+    def as_search_document(self, *, index: str) -> dict:
+        if index == "reservation_units":
+            return {
+                "pk": self.pk,
+                "name_fi": self.name_fi,
+                "name_en": self.name_en,
+                "name_sv": self.name_sv,
+                "description_fi": self.description_fi,
+                "description_en": self.description_en,
+                "description_sv": self.description_sv,
+                "space_name_fi": ",".join([s.name_fi or "" for s in self.spaces.all()]),
+                "space_name_en": ",".join([s.name_en or "" for s in self.spaces.all()]),
+                "space_name_sv": ",".join([s.name_sv or "" for s in self.spaces.all()]),
+                "keyword_groups_name_fi": ",".join(
+                    [k.name_fi or "" for k in self.keyword_groups.all()]
+                ),
+                "keyword_groups_en": ",".join(
+                    [k.name_e or "" for k in self.keyword_groups.all()]
+                ),
+                "keyword_groups_sv": ",".join(
+                    [k.name_sv or "" for k in self.keyword_groups.all()]
+                ),
+                "resources_name_fi": ",".join(
+                    [r.name_fi for r in self.resources.all()]
+                ),
+                "resources_name_en": ",".join(
+                    [r.name_en or "" for r in self.resources.all()]
+                ),
+                "resources_name_sv": ",".join(
+                    [r.name_sv or "" for r in self.resources.all()]
+                ),
+                "services_name_fi": ",".join(
+                    [s.name_fi or "" for s in self.services.all()]
+                ),
+                "services_name_en": ",".join(
+                    [s.name_en or "" for s in self.services.all()]
+                ),
+                "services_name_sv": ",".join(
+                    [s.name_sv or "" for s in self.services.all()]
+                ),
+                "purposes_name_fi": ",".join(
+                    [p.name_fi or "" for p in self.purposes.all()]
+                ),
+                "purposes_name_en": ",".join(
+                    [p.name_en or "" for p in self.purposes.all()]
+                ),
+                "purposes_name_sv": ",".join(
+                    [p.name_sv or "" for p in self.purposes.all()]
+                ),
+                "reservation_unit_type_name_fi": self.reservation_unit_type.name_fi,
```

Comment:
I tried to run the migration on my machine and it failed to this line. `reservation_unit_type` can be `None` so I believe some null checks should be in place here.

Reply:
Excellent catch. I fixed this and also possible unitless reservation unit as well with `getattr`